src/main/java/sc/iview/SciView.java

```diff
@@ -1101,48 +1101,44 @@ public void setColormap( Node n, ColorTable colorTable ) {
+        final NativeTypeEnum nType;
+        if( type instanceof UnsignedByteType ) {
             nType = NativeTypeEnum.UnsignedByte;
-        } else if( voxelType == UnsignedShortType.class ) {
+        } else if( type instanceof UnsignedShortType ) {
             nType = NativeTypeEnum.UnsignedShort;
-        } else if( voxelType == FloatType.class ) {
+        } else if( type instanceof FloatType ) {
             nType = NativeTypeEnum.Float;
         } else {
-            log.debug( "Type: " + voxelType +
+            log.debug( "Type: " + type.getClass() +
                        " cannot be displayed as a volume. Convert to UnsignedByteType, UnsignedShortType, or FloatType." );
             return null;
         }
 
         // Make and populate a ByteBuffer with the content of the Dataset
-        ByteBuffer byteBuffer = ByteBuffer.allocateDirect(
-                ( int ) ( bytesPerVoxel * dimensions[0] * dimensions[1] * dimensions[2] ) );
-        Cursor<T> cursor = image.cursor();
-
-        while( cursor.hasNext() ) {
-            cursor.fwd();
-            if( voxelType == UnsignedByteType.class ) {
-                byteBuffer.put( ( byte ) ( ( ( UnsignedByteType ) cursor.get() ).get() ) );
-            } else if( voxelType == UnsignedShortType.class ) {
-                byteBuffer.putShort( ( short ) Math.abs( ( ( UnsignedShortType ) cursor.get() ).getShort() ) );
-            } else if( voxelType == FloatType.class ) {
-                byteBuffer.putFloat( ( ( FloatType ) cursor.get() ).get() );
-            }
+        final ByteBuffer byteBuffer = ByteBuffer.allocateDirect(
+                ( int ) ( nType.getSizeInBytes() * Intervals.numElements( dimensions ) ) );
+        if( type instanceof UnsignedByteType ) {
+            @SuppressWarnings("unchecked") final IterableInterval< UnsignedByteType > ubytes = ( IterableInterval< UnsignedByteType > ) Views.flatIterable( image );
+            ubytes.forEach( t -> byteBuffer.put( t.getByte() ) );
+        } else if( type instanceof UnsignedShortType ) {
+            @SuppressWarnings("unchecked") final IterableInterval< UnsignedShortType > ushorts = ( IterableInterval< UnsignedShortType > ) Views.flatIterable( image );
+            ushorts.forEach( t -> byteBuffer.putShort( ( t.getShort() ) ) );
+        } else { // if( type instanceof FloatType )
+            @SuppressWarnings("unchecked") final IterableInterval< FloatType > floats = ( IterableInterval< FloatType > ) Views.flatIterable( image );
+            floats.forEach( t -> byteBuffer.putFloat( t.get() ) );
         }
         byteBuffer.flip();
 
         v.readFromBuffer( name, byteBuffer, dimensions[0], dimensions[1], dimensions[2], voxelDimensions[0],
-                          voxelDimensions[1], voxelDimensions[2], nType, bytesPerVoxel );
+                          voxelDimensions[1], voxelDimensions[2], nType, ( int ) nType.getSizeInBytes() );
```

Comment:
Should we assert that `nType.getSizeInBytes() <= Integer.MAX_VALUE` above, just to be safe?

Reply:
No. It's an enum and the maximum that we can get is 8.